app/http.php

```diff
@@ -36,11 +44,53 @@
         // 'document_root' => __DIR__.'/../public',
         // 'enable_static_handler' => true,
         'http_compression' => true,
+        'dispatch_func' => 'dispatch',
         'http_compression_level' => 6,
         'package_max_length' => $payloadSize,
         'buffer_output_size' => $payloadSize,
     ]);
 
+function dispatch(Server $server, $fd, $type, $data = null)
+{
+    global $workerNumber, $domains;
+
+    $safeThreadsPercent = intval(App::getEnv('_APP_SAFE_THREADS_PERCENT', 80)) / 100;
+
+    $safeThreads = (int) floor($workerNumber * $safeThreadsPercent);
+
+    $risky = false;
+    if ($data && str_contains($data, 'POST') && str_contains($data, '/executions')) {
+        $risky = true;
+    } elseif ($data && str_contains($data, '.appwrite.global')) {
+        $risky = true;
+    } elseif ($data) {
+        $lines = explode($data, '\n');
+        if (count($lines) > 2) {
+            $host = trim(explode('Host: ', $lines[1])[1]);
```

Comment:
Hmm, what's most trust-worthy? Host header, or domain parsed from first line (method+url)? Maybe we can use Domains library to simplify the work.

Reply:
First line only has path `/v1` and second line has the host

Reply:
I see.